homeassistant/components/sensibo/climate.py

```diff
@@ -102,244 +93,251 @@ async def async_setup_entry(
-            )
-            self._temperatures_list = (
-                current_capabilities["temperatures"]
-                .get(temperature_unit_key, {})
-                .get("values", [])
-            )
-        else:
-            self._temperature_unit = self._units
-            self._temperatures_list = []
-        self._attr_min_temp = (
-            self._temperatures_list[0] if self._temperatures_list else super().min_temp
+    def get_features(self) -> int:
+        """Get supported features."""
+        features = 0
+        for key in self.coordinator.data[self.unique_id]["features"]:
+            if key in FIELD_TO_FLAG:
+                features |= FIELD_TO_FLAG[key]
+        return features
+
+    @property
+    def current_humidity(self) -> int:
+        """Return the current humidity."""
+        return self.coordinator.data[self.unique_id]["humidity"]
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation."""
+        return (
+            SENSIBO_TO_HA[self.coordinator.data[self.unique_id]["hvac_mode"]]
+            if self.coordinator.data[self.unique_id]["on"]
+            else HVAC_MODE_OFF
         )
-        self._attr_max_temp = (
-            self._temperatures_list[-1] if self._temperatures_list else super().max_temp
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return the list of available hvac operation modes."""
+        return [
+            SENSIBO_TO_HA[mode]
+            for mode in self.coordinator.data[self.unique_id]["hvac_modes"]
+        ]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return self.coordinator.data[self.unique_id]["temp"]
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        return self.coordinator.data[self.unique_id]["target_temp"]
+
+    @property
+    def target_temperature_step(self) -> float | None:
+        """Return the supported step of target temperature."""
+        return self.coordinator.data[self.unique_id]["temp_step"]
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting."""
+        return self.coordinator.data[self.unique_id]["fan_mode"]
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the list of available fan modes."""
+        return self.coordinator.data[self.unique_id]["fan_modes"]
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return the swing setting."""
+        return self.coordinator.data[self.unique_id]["swing_mode"]
+
+    @property
+    def swing_modes(self) -> list[str] | None:
+        """Return the list of available swing modes."""
+        return self.coordinator.data[self.unique_id]["swing_modes"]
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return convert_temperature(
```

Comment:
We don't need to convert here as long as the minimum temperature reported by the device is reported in the same unit as `self.temperature_unit`.

Reply:
Probably will need to address this celsius vs. fahrenheit in next PR but at the moment all these values always comes back as celsius.